packages/gatsby/src/utils/handle-flags.ts

```diff
@@ -69,6 +69,15 @@ const handleFlags = (
   const lockedInFlags = new Map<string, IFlag>()
   const lockedInFlagsThatAreInConfig = new Map<string, IFlag>()
   availableFlags.forEach(flag => {
+    const isForCommand =
+      flag.command === `all` || flag.command === executingCommand
+    // If we're in CI, filter out any flags that don't want to be enabled in CI
+    const isForCi = isCI() ? flag.noCI !== true : true
```

Comment:
Nit: a bit hard to follow. So if `isCI()` is `false` then `isForCi` is always `true`?

Reply:
True

TBH this was copypasta from few lines below https://github.com/gatsbyjs/gatsby/blob/06687500de521d8ce8710a96ec817f87430f80a0/packages/gatsby/src/utils/handle-flags.ts#L96-L100

But I'll think of making it easier to read

Reply:
I tried to make it easier to read - not sure if I succeeded - https://github.com/gatsbyjs/gatsby/pull/30977/commits/35086b3d2b236230340881dc4546106d3eae9269